api/app/src/routes/medical/document.ts

```diff
@@ -65,30 +68,21 @@ router.post(
  *
  * Downloads the specified document for the specified patient.
  *
- * @param req.query.patientId Patient ID for which to retrieve document metadata.
- * @param req.query.facilityId The facility providing NPI for the document download.
- * @param req.query.location The document URL.
- * @param [req.query.mimeType] The mime type of the document.
- * @param [req.query.fileName] The file name of the document.
- * @return The document content.
+ * @param req.query.fileName The file name of the document in s3.
+ * @return presigned url
  */
 router.get(
   "/download",
   asyncHandler(async (req: Request, res: Response) => {
-    const cxId = getCxIdOrFail(req);
-    const patientId = getFromQueryOrFail("patientId", req);
-    const facilityId = getFromQueryOrFail("facilityId", req);
-
-    const location = getFromQueryOrFail("location", req);
-    const mimeType = getFromQuery("mimeType", req);
     const fileName = getFromQuery("fileName", req);
 
-    fileName && res.attachment(fileName);
-    mimeType && res.header("Content-Type", mimeType);
-
-    await downloadDocument({ patientId, cxId, facilityId, location, stream: res });
+    const url = await s3client.getSignedUrl("getObject", {
+      Bucket: Config.getMedicalDocumentsBucketName(),
+      Key: fileName,
+      Expires: 100,
+    });
```

Comment:
this seems like a security issue -> with this approach AFAIK there is nothing stopping a cx guessing file names, and downloading another cx's document

we will need to make this somehow tied to the individual cx only